core/src/main/scala/org/ensime/core/Canon.scala

```diff
@@ -22,7 +21,8 @@ object Canon extends Poly1 {
   // people extend File, so we have to handle subtypes
   implicit def caseFile[F <: File]: Case[F] { type Result = File } = at[F](f => f.canon)
 
-  private val legacyJarUrls: Boolean = sys.props.getOrElse("ensime.legacy.jarurls", "true").toBoolean
+  var serverConfig: EnsimeServerConfig = null
+  private val legacyJarUrls: Boolean = serverConfig.legacyJarUrls
```

Comment:
indeed, should be a `def` because `serverConfig` should be set before `Canon` is accessed.